halo2-ecc/src/fields/fp.rs

```diff
@@ -124,13 +124,46 @@ impl<'range, F: PrimeField, Fp: PrimeField> FpChip<'range, F, Fp> {
         self.gate().assert_is_const(ctx, &borrow.unwrap(), &F::one());
     }
 
+    /// Given proper CRT integer `a`, returns 1 iff `a < modulus::<F>()`
+    /// # Assumptions
+    /// * `a` is proper representation of BigUint
+    pub fn is_less_than_p(
+        &self,
+        ctx: &mut Context<F>,
+        a: impl Into<ProperCrtUint<F>>,
+    ) -> AssignedValue<F> {
+        let a = a.into();
+
+        // underflow != 0 iff carry < p
+        let p = self.load_constant_uint(ctx, self.p.to_biguint().unwrap());
```

Comment:
Just FYI: the enforce_less_than is slightly more optimized because it assigns the constant p limbs while doing the limb subtraction. But it doesn't make much different and your version is cleaner so we can keep it.